lib/jobs/ucs-service-profile-discovery.js

```diff
@@ -80,9 +80,14 @@ function UcsServiceProfileDiscoveryJobFactory(
         
         return self.getRoot()
             .then(function(root) {
-                return [ self.createServiceProfile(root) ];
+                return self.createServiceProfile(root);
             })
-            .spread(function() {
+            .then(function(logicalServers) {
+                self.context.logicalNodeList = (self.context.logicalNodeList || []).concat(
+                    _.map(logicalServers, function(it) {
```

Comment:
@geoff-reid You can use the `_.property` iteratee shorthand for this: `_.map(logicalServers, 'id')`.